File: TUF/String/CalculateFrequencyCharacters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function calculateFrequency(str) {
    var map = new Map();
    for (var i = 0; i < str.length; i++) {
        var currChar = str.charAt(i);
        map.set(currChar, (map.get(currChar) || 0) + 1);
    }
    for (var _i = 0, _a = map.entries(); _i < _a.length; _i++) {
        var _b = _a[_i], Key = _b[0], value = _b[1];
        console.log("".concat(Key, ": ").concat(value));
    }
}
var str = "takeuforward";
calculateFrequency(str);

File: TUF/String/CalculateFrequencyCharacters.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class CalculateFrequencyCharacters {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String str = input.next();
        input.close();
        calculateFrequency(str);
    }

    private static void calculateFrequency(String str) {
        Map<Character, Integer> map = new HashMap<>();
        for (int i = 0; i < str.length(); i++) {
            char currChar = str.charAt(i);
            map.put(currChar, map.getOrDefault(currChar, 0) + 1);
        }
        for (Map.Entry<Character, Integer> entry : map.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }
    }
}
